stdlib/unittest/case.pyi

```diff
@@ -27,6 +27,23 @@ DIFF_OMITTED: str
 class _BaseTestCaseContext:
     def __init__(self, test_case: TestCase) -> None: ...
 
+class _AssertRaisesBaseContext(_BaseTestCaseContext):
+    expected: type[BaseException] | tuple[type[BaseException], ...]
+    expected_regex: str | Pattern[str] | None
+    obj_name: str | None
+    msg: str | None
+
+    def __init__(
+        self,
+        expected: type[BaseException] | tuple[type[BaseException], ...],
+        test_case: TestCase,
+        expected_regex: str | Pattern[str] | None = None,
+    ) -> None: ...
+
+    # This returns Self if args is the empty list, and None otherwise.
+    # but it's not possible to construct an overload which expresses that
+    def handle(self, name: str, args: list[Any], kwargs: dict[str, Any]) -> Any: ...
```

Comment:
I'm about 90% sure this comment is accurate, but if I'm wrong and there's a way to do this overload, I'd be glad to hear it.